Google.Api.Gax/VersionHeaderBuilder.cs

```diff
@@ -103,16 +109,15 @@ private static string GetEntryAssemblyVersionOrNull()
                 return null;
             }
         }
-
-
+        
         private static string FormatAssemblyVersion(System.Type type)
         {
             // Prefer AssemblyInformationalVersion, then AssemblyFileVersion,
             // then AssemblyVersion.
 
             var assembly = type.GetTypeInfo().Assembly;
             var info = assembly.GetCustomAttributes<AssemblyInformationalVersionAttribute>().FirstOrDefault()?.InformationalVersion;
-            if (info != null)
+            if (info != null && IsHeaderNameValueValid(info)) // Skip informational version if it's not a valid header value
```

Comment:
I assume this is because you've seen this going wrong somewhere? Out of interest, what happened?

Reply:
The informational version for `System.Net.Http` is `4.6.30015.01 @BuiltBy: AzDevOps-vseng20170000ZL @Branch: release/2.1-MSRC @SrcCode: https://github.com/dotnet/corefx/tree/96d2fdd8610e6eead9b357789266be9d149dd0b8`
So the code would take it and immediately throw when checking it for spaces and slashes.

Reply:
Ick. That's horrible.